python/ray/tune/trial_runner.py

```diff
@@ -234,13 +235,16 @@ def _process_events(self):
                 result, terminate=(decision == TrialScheduler.STOP))
 
             if decision == TrialScheduler.CONTINUE:
-                if trial.should_checkpoint():
+                if trial.should_checkpoint(result):
                     # TODO(rliaw): This is a blocking call
                     self.trial_executor.save(trial)
                 self.trial_executor.continue_training(trial)
             elif decision == TrialScheduler.PAUSE:
                 self.trial_executor.pause_trial(trial)
             elif decision == TrialScheduler.STOP:
+                # Checkpoint the state before ending the trial if checkpoint_at_end experiment option is set to True
```

Comment:
you may want to run `flake8 --ignore=C408,E121,E123,E126,E226,E24,E704,W503,W504` on this file